engineering-plans/0005-subgraph-composition-query-execution.md

```diff
@@ -0,0 +1,123 @@
+# PLAN-0005: Subgraph Composition Query Execution
+
+<dl>
+  <dt>Author</dt>
+  <dd>Jorge Olivero</dd>
+
+  <dt>Implements</dt>
+  <dd><a href="../rfcs/0001-subgraph-composition.md">RFC-0001 Subgraph Composition</a></dd>
+
+  <dt>Engineering Plan pull request</dt>
+  <dd><a href="">Engineering Plan PR</a></dd>
+
+  <dt>Obsoletes (if applicable)</dt>
+  <dd>-</dd>
+
+  <dt>Date of submission</dt>
+  <dd>2020-1-14</dd>
+
+  <dt>Date of approval</dt>
+  <dd>TBD</dd>
+
+  <dt>Approved by</dt>
+  <dd>TBD</dd>
+</dl>
+
+## Summary
+
+Since subgraph composition allows a subgraph to import types from another subgraph, GraphQL query execution will need to traverse a subgraph's import graph to resolve concrete types.
+Because imported schemas/types are loosely coupled, query execution also needs to address missing types in the api schema during resolution.
+In addition, imported types which have been renamed need to be mapped back to their original name so that queries can resolve correctly.
+Lastly, to simplify query execution graph-node needs to deny subgraphs with JSONB storage from using subgraph composition.
+
+## Implementation
+
+1. Support Custom Type Names: Ensure @originalName directives on the imported types are used to query the correct sql tables.
+2. Codify New Query Runtime Errors: Ensure @placeholder types are acknowledged and proper errors are returned during query execution when a type's subgraph is not deployed to the graph-node resolving the query.
+3. Query Correct Subgraph: Ensure that @subgraphId directives on imported types are used correctly in all cases.
+4. Add Subgraph Composition Feature Flag: Ensure subgraphs JSONB storage can not use subgraph composition.
```

Comment:
I think this isn't really a feature flag. Since no new subgraphs will be deployed using the JSONB schema and composition isn't yet possible, I think an `unreachable!()` in the right place may suffice.